multicluster/controllers/multicluster/member/local_mc_resources_cleanup.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2023 Antrea Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package member
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	k8smcsv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	mcv1alpha1 "antrea.io/antrea/multicluster/apis/multicluster/v1alpha1"
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+	"antrea.io/antrea/pkg/apis/crd/v1beta1"
+)
+
+func cleanUpResourcesCreatedByMC(ctx context.Context, mgrClient client.Client) error {
```

Comment:
We need the same for a leader cluster too?

Also thinking should we prevent removing the leader cluster ClusterSet when there is still at least one `MemberClusterAnnounce` in the leader.

Reply:
I am not sure if it's necessary to clean up all resources in the leader when the ClusterSet is deleted. If the user is just trying to recreate the ClusterSet, and we delete all resources when the ClusterSet CR is deleted, it will require all member clusters to restart to create corresponding ResourceExports again.
If user is trying to clean up the ClusterSet deployment, all resources can be cleaned up after running `kubectl delete -f`
I will add a webhook to warn user when they are trying to delete a ClusterSet CR with at least one `MemberClusterAnnounce` in the leader.

Reply:
I feel no reason users should recreate a ClusterSet by deleting the old ClusterSet CR and create a new one. What is the purpose of that? `kubectl delete -f` wont clean up auto-created CRs right.

We can disallow ClusterSet deletion when there is a least one `MemberClusterAnnounce`, but how about resources created due to changes in the leader cluster (not due to member cluster export), like ANPImport?

Reply:
Yeah, I think we can assume that user is trying to clean up a ClusterSet by deleting a ClusterSet CR. Recreate shouldn't be a common case.

`kubectl delete -f` wont clean up auto-created CRs, I am still trying to figure out how to clean up resources when user run `kubectl delete -f`, we may use a guide and antctl.

I am not sure I get your second question, do you mean how to handle the deletion of ANPImport in the leader cluster?

Reply:
I meant ANPImport is one example of auto-generated CRs that ideally we should delete when ClusterSet is deleted.

Reply:
Ok, yes, I am working on adding more codes to clean up all auto-created resources when the ClusterSet is deleted in a leader cluster.

Reply:
Yeah, at least resources created for changes from the leader cluster itself. For other resource export CRs created for member exports, let us see how complex to auto-delete them, or we just disallow ClusterSet deletion when there are still MemberAnnounces.

Reply:
Added a webhook to stop delete ClusterSet CR on the leader if there is any remaining MemberClusterAnnounces. #5475.
Refine the ClusterSet CR deletion event handling part in the leader ClusterSet controller to clean up all remaining ResourceExports. (Basically, there should be ACNP type of ResourceExports only to be deleted if we stop user to delete a ClusterSet CR in the leader cluster)
Update the user guide to guide user about how to remove a member or a leader cluster.